src/transport_ofi.c

```diff
@@ -1624,6 +1624,8 @@ int shmem_transport_fini(void)
 
     for (i = 0; i < shmem_transport_ofi_contexts_len; ++i) {
         if (shmem_transport_ofi_contexts[i]) {
+            if (shmem_transport_ofi_contexts[i]->options & SHMEM_CTX_PRIVATE)
+                RAISE_WARN_MSG("Shutting down with unfreed private context (%zd)\n", i);
```

Comment:
@davidozog Note that we are entering private contexts into the contexts array, so this loop will free them.  As a result the check below on line 1646 should really only trip if something went wrong with STX management.  I added this warning message to ensure the user gets a warning about unfreed private contexts.  Not sure if you want to make any broader changes with regard to how private contexts are handled.  At a minimum, maybe we want to update the warning message below?

Reply:
Yup, this looks good to me.  And I agree that 1646 should never happen unless something strange occurs.  Should we just remove it?  I'm not sure what we would print other than something like "unexpected context(s) was not freed".

Reply:
Looks like past Jim also filed an issue for this: #579